hazelcast/src/main/java/com/hazelcast/map/impl/AbstractEvictableRecordStore.java

```diff
@@ -382,6 +383,16 @@ protected void accessRecord(Record record, long now) {
         setExpirationTime(record, maxIdleMillis);
     }
 
+    protected void mergeRecordExpiration(Record record, EntryView mergingEntry) {
+        if (mergingEntry.getTtl() > 0 && mergingEntry.getTtl() != Long.MAX_VALUE) {
+            record.setTtl(mergingEntry.getTtl());
+        }
+        if (mergingEntry.getExpirationTime() > 0 && mergingEntry.getExpirationTime() != Long.MAX_VALUE) {
+            record.getStatistics().setExpirationTime(mergingEntry.getExpirationTime());
```

Comment:
Here `record.getStatistics()` may throw NPE.
 
To set expiration time correctly you should call this method `ExpirationTimeSetter#setExpirationTime` that method takes into account `max-idle-seconds` and also nullability of `record.getStatistics()`.